File: tessoku/src/A12.py
n, k = map(int, input().split())
a = list(map(int, input().split()))

l = 0
r = 10**9

while l + 1 != r:
    m = (l + r) // 2
    cnt = sum(m // x for x in a)

    if cnt < k:
        l = m
    else:
        r = m

print(r)

File: tessoku/rust/src/bin/A12.rs
use proconio::input;

fn check(a: &Vec<usize>, x: usize, k: usize) -> bool {
    let mut sum = 0;
    for i in 0..a.len() {
        sum += x / a[i]
    }

    sum >= k
}

fn main() {
    input! {
        n: usize,
        k: usize,
        a: [usize; n],
    }

    let mut l = 0;
    let mut r = 100_000_000;
    while l < r {
        let m = (l + r) / 2;
        let ans = check(&a, m, k);
        if ans {
            r = m
        } else {
            l = m + 1;
        }
    }

    println!("{}", l);
}
